util/test/Test/Pos/Util/Golden.hs

```diff
@@ -38,6 +43,35 @@ goldenTestJSON x path = withFrozenCallStack $ do
             Left err -> failWith Nothing $ "could not decode: " <> show err
             Right x' -> x === x'
 
+-- | Only check that the datatype equals the decoding of the file
+goldenTestJSONDec :: (Eq a, FromJSON a, HasCallStack, Show a)
+                  => a -> FilePath -> Property
+goldenTestJSONDec x path = withFrozenCallStack $ do
+    withTests 1 . property $ do
+        bs <- liftIO (LB.readFile path)
+        case eitherDecode bs of
+            Left err -> failWith Nothing $ "could not decode: " <> show err
+            Right x' -> x === x'
+
+goldenTestCanonicalJSONDec
+    :: ( Eq a
+       , Canonical.FromJSON (Either SchemaError) a
+       , HasCallStack
+       , Show a
+       )
+    => a
+    -> FilePath
+    -> Property
+goldenTestCanonicalJSONDec x path = withFrozenCallStack $ do
```

Comment:
Same question as above.

Reply:
See above comment. These should be used sparingly, only to ensure backwards compatibility.